### PR TITLE
RavenDB-16768: fix failing test, don't wait for WaitForExecutionOnSpecificNode on single node

### PR DESCRIPTION
**Pull request by garayx**

### CODE REVIEW

File: src/Raven.Server/Documents/Handlers/SubscriptionsHandler.cs [hunk 1]
```diff
@@ -499,11 +499,15 @@ private async Task CreateInternal(BlittableJsonReaderObject bjro, SubscriptionCr
             using (ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext serverContext))
             using (serverContext.OpenReadTransaction())
             {
-                // need to wait on the relevant remote node
-                var node = Database.SubscriptionStorage.GetResponsibleNode(serverContext, name);
-                if (node != null && node != ServerStore.NodeTag)
+                var topology = ServerStore.GetClusterTopology(serverContext);
+                if (topology.AllNodes.Count != 1)
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Handlers/SubscriptionsHandler.cs]: Shouldn't we use Members here? cc @karmeli87

File: src/Raven.Server/Documents/Handlers/SubscriptionsHandler.cs [hunk 2]
```diff
@@ -499,11 +499,15 @@ private async Task CreateInternal(BlittableJsonReaderObject bjro, SubscriptionCr
...
                 {
-                    await WaitForExecutionOnSpecificNode(serverContext, ServerStore.GetClusterTopology(serverContext), node, subscriptionId);
+                    // need to wait on the relevant remote node
+                    var node = Database.SubscriptionStorage.GetResponsibleNode(serverContext, name);
+                    if (node != null && node != ServerStore.NodeTag)
+                    {
+                        await WaitForExecutionOnSpecificNode(serverContext, topology, node, subscriptionId);
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Handlers/SubscriptionsHandler.cs]: Don't we need at least to execute `await ServerStore.Cluster.WaitForIndexNotification(index);` ?

Comment (garayx) [reply src/Raven.Server/Documents/Handlers/SubscriptionsHandler.cs]: no if its single node we waited inside PutSubscription

File: src/Raven.Server/Documents/Handlers/SubscriptionsHandler.cs (hunk 1 above)
Comment (garayx) [reply src/Raven.Server/Documents/Handlers/SubscriptionsHandler.cs]: this check is only to know if its single node cluster

Comment (karmeli87) [reply src/Raven.Server/Documents/Handlers/SubscriptionsHandler.cs]: What will happened if you create a database in cluster of 3 nodes, but with replication factor of 1?
node A executes the command, but the database is created on node B.
Would that work as you expect?

Comment (karmeli87) [reply src/Raven.Server/Documents/Handlers/SubscriptionsHandler.cs]: okay... this is the _cluster_ topology.

**Comment by karmeli87** [comment]
I don't understand why this change can fix anything?

**Comment by garayx** [comment]
> I don't understand why this change can fix anything?

1st commit is optimization the actual fix for test is in 2nd commit

**Pull request merged**